### PR TITLE
[main] Upgrade libsolv to v0.7.20, libsoup to v3.0.4, glib to v2.71.0, gtk-doc to v1.33.2

### PR DESCRIPTION
**Pull request by henryli001**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- Upgrade to libsolv to v0.7.20
- Upgrade libsoup to v3.0.4, which requires upgrading glib to 2.71.0, gtk-doc to 1.33.2, vala to 0.54.6, open-vm-tools to 11.3, ostree to 2022.1, rest to 0.9.0, rpm-ostree to 2022.1, vte291 to v0.66.2
-  Update gtk3 to use the upgraded rest as BR

###### Change Log  

1. libsolv:
- Upgrade to version 0.7.20

2. libsoup:
- - Upgrade to version 3.0.4
- Add cmake, libnghttp2-devel, brotli-devel, gnutls-devel, 
  gtk-doc, vala and python3-pygments as BR
- Use meson to build and install
- Add additional files to libsoup-devel 

3. glib:
- Upgrade to version 2.71.0
- Add python3-pygments as BR
- Don't remove pcre sources which no longer apply for the new version

4. gtk-doc:
- Upgrade to version 1.33.2
- Add meson and python3-pygments as BR
- Use meson to build and install
- Disable test, which will cause circular dependency with glib2

5. vala: 
- Upgrade to version 0.54.6
- Add bison, flex, gobject-introspection-devel, libxslt and graphviz-devel as BR
- Add subpackages vala-doc, valadoc and valadoc-devel
- Remove /usr/bin/vapicheck* since the files no longer exist
- Remove duplicate %license COPYING from main package 

6. open-vm-tools:
- Upgrade to version 11.3.0
- Remove patches that no longer apply
- Add vmware-alias-import,libgdp.so, libguestStore.so, libguestStoreClient.so
  and libguestStoreClient.so.*

7. ostree:
- Upgrade to version 2022.1
- Fix Source0 field to use macro to represent package version

8. rest:
- Upgrade to version 0.9.0
- Remove patch that no longer applies
- Remove autoconf, automake and gtk-doc as BR
- Add meson, cmake and json-glib-devel as BR
- Disable soup2 to use libsoup 3.0 as dependency
- Disable gtk-doc which will involve unresolved dependencies in Mariner
- Modify the file names due to version change

9. rpm-ostree:
- Upgrade to version 2022.1
- Remove patches that no longer apply
- Fix rpm-ostree-disable-selinux.patch
- Add new files to main package due to the version upgrade

10. gtk3:
- Use pkgconfig(rest-1.0) as BR

11. vte291:
- Upgrade to version 0.66.2
- Update vte291-cntnr-precmd-preexec-scroll.patch
- Update gtk3_version macro
- Update Source0 URL to use macros
- Add %{_datadir}/glade/ to vte291-devel package
- Update license and doc files

###### Does this affect the toolchain?  

YES

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Local Build

### CODE REVIEW

File: SPECS/glib/glib.spec [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
 URL:            https://developer.gnome.org/glib/
-Source0:        https://ftp.gnome.org/pub/gnome/sources/glib/2.60/%{name}-%{version}.tar.xz
-Patch0:         CVE-2019-12450.patch
-Patch1:         CVE-2020-35457.patch
-# CVE-2021-27218 and CVE-2021-27219 are both solved by the patch for the first
-Patch3:         CVE-2021-27218.patch
-Patch4:         CVE-2021-27219.nopatch
-Patch5:         CVE-2021-28153.patch
+#Source0:        https://ftp.gnome.org/pub/gnome/sources/glib/2.60/%{name}-%{version}.tar.xz
+Source0:        %{name}-%{version}.tar.xz
```
Comment (PawelWMS) [inline SPECS/glib/glib.spec:11]: I think the version in the comment is outdated. Also, why not

```suggestion
Source0:        https://ftp.gnome.org/pub/gnome/sources/glib/%{majmin}/%{name}-%{version}.tar.xz
```

with an additional

```bash
%define majmin %(echo %{version} | cut -d. -f1-2)
```

at the top of the spec?

File: SPECS/graphviz/graphviz.spec [hunk 2]
```diff
@@ -51,7 +51,7 @@ BuildRequires:		python2-devel
 %endif
 BuildRequires:		python3-devel, java-devel
 BuildRequires:		pkgconfig(cairo) >= 1.1.10, pango-devel, gmp-devel, lua-devel
-BuildRequires:		gd-devel, perl-devel, swig >= 1.3.33, automake, autoconf, libtool
+BuildRequires:		gd-devel, perl-devel, perl, swig >= 1.3.33, automake, autoconf, libtool
```
Comment (PawelWMS) [inline SPECS/graphviz/graphviz.spec:54]: Didn't you migrate `graphviz` into `SPECS` yesterday?

File: SPECS/gtk-doc/gtk-doc.spec [hunk 3]
```diff
@@ -32,13 +36,20 @@ BuildArch:      noarch
...
 %prep
 %setup -q
+
+# Move this doc file to avoid name collisions	
+mv doc/README doc/README.docs
+
 %build
-%configure --disable-silent-rules CFLAGS="%{build_cflags}"
-make %{?_smp_mflags}
+%meson -Dtests=false
```
Comment (PawelWMS) [inline SPECS/gtk-doc/gtk-doc.spec:47]: Could you also check if we don't need to update the `%check` section as well now?

File: SPECS/libpq/libpq.signatures.json [hunk 4]
```diff
@@ -1,6 +1,5 @@
 {
  "Signatures": {
-  "postgresql-12.2.tar.bz2": "ad1dcc4c4fc500786b745635a9e1eba950195ce20b8913f50345bb7d5369b5de",
-  "postgresql-12.2.tar.bz2.sha256": "340ec030e195388b900fb06f0d8b90de48de6ca4e9d5df0946d18d966cb40798"
```
Comment (PawelWMS) [inline SPECS/libpq/libpq.signatures.json:4]: Good catch!

File: SPECS/libsoup/libsoup.spec [hunk 5]
```diff
@@ -61,22 +65,23 @@ Requires:       %{name} = %{version}-%{release}
...
 
 %install
-rm -rf %{buildroot}%{_infodir}
-make DESTDIR=%{buildroot} install
+%meson_install
 find %{buildroot} -type f -name "*.la" -delete -print
 
-%find_lang %{name}
-
-%check
```
Comment (PawelWMS) [inline SPECS/libsoup/libsoup.spec:78]: Why did we need to remove the `%check` section?

File: SPECS/open-vm-tools/open-vm-tools.spec [hunk 6]
```diff
@@ -39,9 +39,6 @@ Source2:        %{vgauthdaemon}.service
 Source3:        vmblock.mount
 Source4:        open-vm-tools.conf
 Source5:        vmtoolsd.pam
-Patch1:         gcc10-warning.patch
-# Cumulative patch for fixes from https://github.com/vmware/open-vm-tools/tree/stable-11.1.0-SDMP-fixes
-Patch2:         sdmp-fixes.patch
```
Comment (PawelWMS) [inline SPECS/open-vm-tools/open-vm-tools.spec:44]: I think the files still need to be removed.

File: SPECS/glib/glib.spec (hunk 1 above)
Comment (henryli001) [reply SPECS/glib/glib.spec]: Fixed

File: SPECS/libsoup/libsoup.spec (hunk 5 above)
Comment (henryli001) [reply SPECS/libsoup/libsoup.spec:82]: Added back

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**